keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
 
-This call grants access to a particular principal. The principal is the account for which this access should be granted. 
+## COSI Driver gRPC API
 
-If the principal is set, then it should be used as the username of the created credentials or in some way should be deterministically used to generate a new credential for this request. This principal will be used as the unique identifier for deleting this access by calling ProvisionerRevokeBucketAccess
+#### ProvisionerGetInfo
 
-If the `principal` is empty, then a new service account should be created in the backend that satisfies the requested `access_policy`. The username/principal for this service account should be set in the `principal` field of `ProvisionerGrantBucketAccessResponse`.
+This gRPC call responds with the name of the provisioner. The name is used to identify the appropriate driver for a given BucketRequest or BucketAccessRequest.
 
 ```
-message ProvisionerGrantBucketAccessRequest {
-    // This field is REQUIRED
-    string bucket_name = 1;
-    
-    map<string,string> bucket_context = 2;  
+    ProvisionerGetInfo                                           
+    |------------------------------------------|       |---------------------------------------|
+    | grpc ProvisionerGetInfoRequest{}         | ===>  | ProvisionerGetInfoResponse{           | 
+    |------------------------------------------|       |   "name": "s3.amazonaws.com"          |
+                                                       | }                                     |
+                                                       |---------------------------------------|
+```
 
-    string principal = 3;
-    
-    // This field is REQUIRED
-    string access_policy = 4;
-}
+#### ProvisionerCreateBucket
 
-message ProvisionerGrantBucketAccessResponse {
-    // This is the account that is being provided access. This will
-    // be required later to revoke access. 
-    string principal = 1;
-    
-    string credentials_file_contents = 2;
-    
-    string credentials_file_path = 3;
-} 
+This gRPC call creates a bucket in the OSP, and returns information about the new bucket. This api must be idempotent. The input to this call is the name of the bucket and an opaque parameters field. 
+
+The returned `bucketID` should be a unique identifier for the bucket in the OSP. This value could be the name of the bucket too. This value will be used by COSI to make all subsequent calls related to this bucket.
+ 
+```
+    ProvisionerCreateBucket
+    |------------------------------------------|       |-----------------------------------------------|
+    | grpc ProvisionerCreateBucketRequest{     | ===>  | ProvisionerCreateBucketResponse{              |
+    |     "name": "br-$uuid",                  |       |   "bucketID": "br-$uuid",                     |
```

Comment:
Is "br-$uuid" supposed to be the COSI-generated name of the OSP bucket or the name of the COSI Bucket instance? I thought the later, though I guess it could be both (I'd still like to see prefix added back to the BR). 
Regardess, the response bucketID should be an OSP-generated unique name, so this is not likely "br-$uuid", right?

Reply:
It could be br-$uuid, however, like you said. It need not be.